shell/platform/linux/fl_value.cc

```diff
@@ -92,6 +92,120 @@ static ssize_t fl_value_lookup_index(FlValue* self, FlValue* key) {
   return -1;
 }
 
+static void int_to_string(GString* buffer, int64_t value) {
+  g_string_append_printf(buffer, "%" G_GINT64_FORMAT, value);
+}
+
+static void float_to_string(GString* buffer, double value) {
+  g_string_append_printf(buffer, "%.16f", value);
+
+  // Strip trailing zeros
+  int zero_count = 0;
+  for (int i = buffer->len - 1; i >= 0; i--) {
+    // Leave one zero after a decimal point
+    if (buffer->str[i] == '.') {
+      zero_count = zero_count == 0 ? 0 : zero_count - 1;
+      break;
+    }
+    if (buffer->str[i] != '0')
+      break;
+    zero_count++;
+  }
+  g_string_truncate(buffer, buffer->len - zero_count);
+}
+
+static void value_to_string(GString* buffer, FlValue* value) {
+  switch (value->type) {
+    case FL_VALUE_TYPE_NULL:
+      g_string_append(buffer, "null");
+      return;
+    case FL_VALUE_TYPE_BOOL:
+      if (fl_value_get_bool(value))
+        g_string_append(buffer, "true");
+      else
+        g_string_append(buffer, "false");
+      return;
+    case FL_VALUE_TYPE_INT:
+      int_to_string(buffer, fl_value_get_int(value));
+      return;
+    case FL_VALUE_TYPE_FLOAT:
+      float_to_string(buffer, fl_value_get_float(value));
+      return;
+    case FL_VALUE_TYPE_STRING: {
+      g_string_append(buffer, fl_value_get_string(value));
+      return;
+    }
+    case FL_VALUE_TYPE_UINT8_LIST: {
+      g_string_append(buffer, "[");
+      const uint8_t* values = fl_value_get_uint8_list(value);
+      for (size_t i = 0; i < fl_value_get_length(value); i++) {
```

Comment:
Optional: you could pull everything from here to the end of the case into a template function, with the value type (e.g., `uint8_t`) as the template type, and the `*_to_string*` function as a function pointer parameter, then share that code for all the base-type lists (so, everything but the `FlValue` list).

Reply:
I'll leave these as is because I don't want to add too much C++ with the C code and have a mixed style.